generators/server/files.js

```diff
@@ -1750,20 +1753,30 @@ function writeFiles() {
         modifyFiles() {
             if (this.buildTool === 'gradle') {
                 this.addGradleProperty('kotlin_version', kotlinConstants.KOTLIN_VERSION);
+                this.addGradleProperty('detekt_version', kotlinConstants.DETEKT_VERSION);
                 this.addGradlePlugin('org.jetbrains.kotlin', 'kotlin-gradle-plugin', '${kotlin_version}');
                 this.addGradlePlugin('org.jetbrains.kotlin', 'kotlin-allopen', '${kotlin_version}');
                 if (this.databaseType === 'sql') {
                     this.addGradlePlugin('org.jetbrains.kotlin', 'kotlin-noarg', '${kotlin_version}');
                 }
                 this.addGradlePlugin('org.jlleitschuh.gradle', 'ktlint-gradle', kotlinConstants.KTLINT_GRADLE_VERSION);
+                this.addGradlePlugin('io.gitlab.arturbosch.detekt', 'detekt-gradle-plugin', '${detekt_version}');
 
                 this.applyFromGradleScript('gradle/kotlin');
             }
 
             if (this.buildTool === 'maven') {
+                this.addMavenPluginRepository('jcenter', 'https://jcenter.bintray.com/');
+
                 this.addMavenProperty('kotlin.version', kotlinConstants.KOTLIN_VERSION);
                 this.addMavenProperty('ktlint.version', kotlinConstants.KTLINT_VERSION);
                 this.addMavenProperty('maven-antrun-plugin.version', kotlinConstants.MAVEN_ANTRUN_VERSION);
+                this.addMavenProperty('detekt.version', kotlinConstants.DETEKT_VERSION);
+                this.addMavenProperty('detekt.configFile', '${project.basedir}/detekt-config.yml');
```

Comment:
can we move this into constants or variables?

Reply:
If I understood correctly you mean the name of the config file. Please see my last commit and let me know if otherwise. Thanks